src/collapse/vsprof.rs

```diff
@@ -0,0 +1,188 @@
+                //    outer function
+                // 2. prev_depth == depth -> the previous function call was a leaf node, so we
+                //    need to save the current stack and replace the top node with our node
+                //    call
+                // 3. prev_depth > depth -> the previous function call was a leaf node, so we
+                //    need to save the current stack and than we need to pop the top nodes
+                //    until the top node is our parent
+                match prev_depth.cmp(&depth) {
+                    // Case 1
+                    Ordering::Less => {
+                        self.stack
+                            .push((function_name.to_string(), number_of_calls));
+                    }
+                    // Case 2
+                    Ordering::Equal => {
+                        self.write_stack(occurences);
+                        self.stack.pop();
+                        self.stack
+                            .push((function_name.to_string(), number_of_calls));
+                    }
+                    // Case 3
+                    Ordering::Greater => {
+                        self.write_stack(occurences);
+
+                        // Pop (prev_depth - depth - 1) nodes of the stack, this leaves our
+                        // parent as the top node.
+                        for _ in 0..=prev_depth - depth {
+                            self.stack.pop();
+                        }
+
+                        self.stack
+                            .push((function_name.to_string(), number_of_calls));
+                    }
+                }
+            }
+            _ => {
+                return invalid_data_error!("Unable to parse function name from line:\n{}", line);
+            }
+        }
+
+        Ok(())
+    }
+
+    // Store the current stack in `occurences`
+    fn write_stack(&self, occurences: &mut Occurrences) {
+        let (functions, samples): (Vec<_>, Vec<_>) = self.stack.iter().cloned().unzip();
+        if let Some(number_of_calls) = samples.last() {
+            if *number_of_calls > 0 {
+                occurences.insert(functions.join(";"), *number_of_calls);
+            }
+        }
+    }
+}
+
+/// Gets the number from the start of the line. This can either be a number <1000, in which case the
+/// line doesn't contain double quotes, or the number can be >1000, in which case the line does
+/// contain double quotes. In both cases `line` may start with a leading comma, which will be
+/// ignored.
+fn get_next_number(line: &str) -> io::Result<(usize, &str)> {
+    // Trim the leading comma, if any
+    let line = line.trim_start_matches(',');
+
+    if let Some('"') = line.chars().next() {
+        // number >1000
+        // Trim the leading double quote and then split on the next double quote, which leaves
+        // the number as the first part of the result.
+        let mut split = line.trim_start_matches('"').splitn(2, '"');
```

Comment:
```suggestion
        let split = rest.split_once('"');
```
Crucially, this will return `None` if there is no second `"` in the string, whereas `.splitn` will return the remainder of the string if it contains no `"`.

Reply:
I was using `split_once` originally, which was much more convenient, but it has only been stabilized in `1.52.0`, so this made the CI check for compilation on `1.46.0` fail. Unless the minimum compiler version for this project is changed to `1.52.0`, I think this isn't possible.

Reply:
I'm okay with bumping to 1.52.0. It was released May 6th, which is now 5 months ago. Assuming the next release of inferno with this change goes out in, say, a month, that's a 6 month window, which I'm comfortable with.

Reply:
Should I bump the version in this PR, or would you prefer a separate PR for that? When running clippy I get quite a few warnings (in other files), which may or may not be new since 1.46.0. With a separate PR I feel like that would present a good opportunity to include fixes for those warnings.